lib/polling-consumer.test.js

```diff
@@ -368,4 +371,65 @@ describe('lib/polling-consumer', () => {
     });
     consumer.start();
   });
+
+  test('the consumer keeps polling with a re-created iterator if the previous one expires', done => {
+    getShardIterator.mockResolvedValueOnce({ ShardIterator: 'iterator-1' });
+    getShardIterator.mockResolvedValueOnce({ ShardIterator: 'iterator-3' });
+    getRecords.mockResolvedValueOnce({
+      MillisBehindLatest: 10,
+      NextShardIterator: 'iterator-2',
+      Records: [{ Data: 'foo', SequenceNumber: 1 }]
+    });
+    getRecords.mockRejectedValueOnce(
+      Object.assign(new Error('The iterator has expired'), { code: 'ExpiredIteratorException' })
+    );
+    const consumer = new PollingConsumer(options);
+    setTimeout.mockImplementationOnce((callback, delay, ...args) => {
```

Comment:
Why not use fake timers instead?

Reply:
We are, with custom implementations of the mock.